docs/reference/cro-webapp-template-syntax.md

```diff
@@ -396,6 +396,55 @@ be used as:
 
 To set the current target for the body in a macro, use `<:body $target>`.
 
+## Fragments
+
+A template fragment works somewhat like a template subroutine, except that it is
+intended to be placed inline within other template content. In this example, 
+the fragment is used with HTMX tags (see https://htmx.org) to provide dynamic 
+behaviour.
+
+```
+<html>
+  <body>
+    <div hx-target="this">
+      <:fragment archive-ui($_)>
+        <?.archived>
+        <button hx-patch="/contacts/<.id>/unarchive">Unarchive</button>
+        </?>
+        <!>
+        <button hx-delete="/contacts/<.id>">Archive</button>
+        </!>
+      </:>
+    </div>
+    <h3>Contact</h3>
+    <p><.email></p>
+  </body>
+</html>
+```
+
+The entire template can be rendered as normal, in which case the `<:fragment...>` 
+and `</:>` delimiters are ignored. Template content inside the fragment delimiters 
+can contain conditionals, variables and so on which use the outer topic.
+
+Alternatively, the fragment can be rendered individually from a route block 
+by specifying its name:
+
+```
+template('templates/page.crotmp', :fragment<archive-ui>, 
+            { archived => True, id => 42, email => 'me@me.com' }));
+```
+
+A fragment should be written with parameter of `$_` to set its internal topic.
+
+Or, the fragment can be rendered individually from another template like a 
+regular sub using `§` in place of `&`:
```

Comment:
OK

However, I think that `#` is used for crotmp comments.

FWIW I think that the classical use for fragments is direct from the route block so likely 80-90% of uses will not call the fragment from another template file. However, I kept a way for another template file to call a fragment analogous to `sub` operation since I think that there may be some more involved use cases where fragments and sub may be "outsourced" to eg a shared library template.

I will change to tilde `~` here and in the cro-webapp code review